cmds/shutdown/shutdown.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// shutdown halts or reboots.
+//
+// Synopsis:
+//     shutdown [-dryrun] [operation]
+//
+// Description:
+//     shutdown will either do or simulate the operation.
+//     current operations are reboot and halt.
+//
+// Options:
+//     -dryrun:   do not do really do it.
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+	"syscall"
+)
+
+var (
+	dryrun  = flag.Bool("dryrun", false, "Do not do kexec system calls")
+	op      = "reboot"
+	opcodes = map[string]uintptr{
+		"halt":    syscall.LINUX_REBOOT_CMD_POWER_OFF,
+		"reboot":  syscall.LINUX_REBOOT_CMD_RESTART,
+		"suspend": syscall.LINUX_REBOOT_CMD_SW_SUSPEND,
+	}
+)
+
+func usage() {
+	log.Fatalf("shutdown [-dryrun] [halt|reboot|suspend] (defaults to reboot)")
```

Comment:
are we not following the gnu shutdown command format? it's a little different. 

 shutdown [OPTION]...  TIME [MESSAGE]

It's ok if we don't want to. I just wanted to check